spikeinterface/sorters/launcher.py

```diff
@@ -46,6 +44,111 @@ def _run_one(arg_list):
 _implemented_engine = ('loop', 'joblib', 'dask')
 
 
+def run_sorter_by_property(sorter_name,
+                           recording,
+                           grouping_property,
+                           working_folder,
+                           mode_if_folder_exists='raise',
+                           engine='loop',
+                           engine_kwargs={},
+                           verbose=False,
+                           docker_image=None,
+                           **sorter_params):
+    """
+    Generic function to run a sorter on a recording after splitting by a 'goruping_property' (e.g. 'group').
+
+    Internally, the function works as follows:
+        * the recording is split based on the provided 'grouping_property' (using the 'split_by' function)
+        * the 'run_sorters' function is run on the split recordings
+        * sorting outputs are aggregated using the 'aggregate_units' function
+        * the 'grouping_property' is added as a property to the SortingExtractor
+
+    Parameters
+    ----------
+
+    sorter_name: str
+        The sorter name
+
+    recording: BaseRecording
+        The recording to be sorted
+
+    grouping_property: object
+        Property to split by before sorting
+
+    working_folder: str
+        The working directory.
+
+    sorter_params: dict of dict with sorter_name as key
+        This allow to overwrite default params for sorter.
+
+    mode_if_folder_exists: 'raise_if_exists' or 'overwrite' or 'keep'
+        The mode when the subfolder of recording/sorter already exists.
+            * 'raise' : raise error if subfolder exists
+            * 'overwrite' : delete and force recompute
+            * 'keep' : do not compute again if f=subfolder exists and log is OK
+
+    engine: str
+        'loop', 'joblib', or 'dask'
+
+    engine_kwargs: dict
+        This contains kwargs specific to the launcher engine:
+            * 'loop' : no kwargs
+            * 'joblib' : {'n_jobs' : } number of processes
+            * 'dask' : {'client':} the dask client for submiting task
+
+    verbose: bool
+        default True
```

Comment:
Same here. Avioid copy/paste for documentation but if if change at one place we always forgot to change the other place. Copy/paste is a bad pratice.